dbms/src/Interpreters/sortBlock.cpp

```diff
@@ -77,9 +84,9 @@ struct PartialSortingLessWithCollation
         for (ColumnsWithSortDescriptions::const_iterator it = columns.begin(); it != columns.end(); ++it)
         {
             int res;
-            if (needCollation(it->first, it->second))
+            if (auto column_string_ptr = needCollation(it->first, it->second))
             {
-                const ColumnString & column_string = typeid_cast<const ColumnString &>(*it->first);
+                const ColumnString & column_string = typeid_cast<const ColumnString &>(*column_string_ptr);
                 res = column_string.compareAtWithCollation(a, b, *it->first, *it->second.collator);
```

Comment:
I think that if `needCollation` have returned nested column, it would have size=1. But we compare rows with numbers `a` and 'b'.

Reply:
I'll write a test.